pkgs/servers/code-server/default.nix

```diff
@@ -166,11 +176,24 @@ in stdenv.mkDerivation rec {
       -execdir ln -s ${ripgrep}/bin/rg {}/bin/rg \;
 
     # patch shebangs of everything, also cached files, as otherwise postinstall
-    # will not be able to find /usr/bin/env, as it does not exists in sandbox
+    # will not be able to find /usr/bin/env, as it does not exist in sandbox
     patchShebangs .
 
-    # rebuild binaries, we use npm here, as yarn does not provider alternative
-    # that would not atempt to try to reinstall everything and break out
+    '' + stdenv.lib.strings.optionalString stdenv.isDarwin ''
+
+    # Playwright is only needed for tests, we can disable it for builds.
+    # There's an environment variable to disable downloads, but the package makes a breaking call to
+    # sw_vers before that variable is checked.
+    patch -p1 -i ${./darwin-playwright.patch}
+
+    # fsevents build fails on Darwin. It's an optional package, but the subsequent rebuild
+    # treats it as mandatory
+    patch -p1 -i ${./darwin-fsevents.patch}
```

Comment:
Can we apply this patch non conditionally?

Reply:
The package is only installed on Darwin systems, so this one fails to patch on NixOS unless it's conditional.

Reply:
Can you add a comment that the file only exists on darwin?

Reply:
Yep, added this